diff --git a/js/govuk.js b/js/govuk.js
index df70c90..f576507 100755
--- a/js/govuk.js
+++ b/js/govuk.js
@@ -1,13 +1,29 @@
 function toggleMobileMenu() {
+
+  // Check if mobile menu toggle is visible to hide the desktop menu from assistive technology
+  if (!jQuery("#mobile-menu-toggle").is(":visible")) {
+    jQuery(".govuk-header__navigation").parent(".app-navigation").attr("aria-hidden","true");
+  }
+  else {
+    jQuery(".govuk-header__navigation").parent(".app-navigation").attr("aria-hidden","false");
+  }
+
+  // Load menu items if menu container is empty, remove the h2 as this will have a duplicate ID
   if (!jQuery("#mobile-menu").html()) {
     let content = jQuery(".govuk-header__navigation").parent(".app-navigation").html();
-    let html = "<div id='mobile-menu-content'>" + content + "</div>";
-    jQuery("#mobile-menu").html(html);
+    let html = "<nav id='mobile-menu-content'>" + content + "</nav>";
+    jQuery("#mobile-menu").html(html).find('h2').remove();
   }
+
   if (!jQuery("#mobile-menu-content").is(":visible")) {
     jQuery("#mobile-menu-content").slideDown();
+    jQuery("#mobile-menu-toggle").attr("aria-expanded","true");
+    jQuery("#mobile-menu").attr("aria-hidden","false");
+    jQuery("#mobile-menu").attr("aria-hidden","false");
   }
   else {
     jQuery("#mobile-menu-content").slideUp();
+    jQuery("#mobile-menu-toggle").attr("aria-expanded","false");
+    jQuery("#mobile-menu").attr("aria-hidden","true");
   }
 }
diff --git a/sass/_mobile-menu.scss b/sass/_mobile-menu.scss
index d321b25..d2429cd 100755
--- a/sass/_mobile-menu.scss
+++ b/sass/_mobile-menu.scss
@@ -18,6 +18,9 @@
   #mobile-menu {
     position: relative;
     width: 100%;
+    @include govuk-media-query($from: desktop) {
+      display: none;
+    }
     #mobile-menu-content {
       display: none;
       position: absolute;
diff --git a/templates/layout/page--404.html.twig b/templates/layout/page--404.html.twig
index dc61b73..290af22 100755
--- a/templates/layout/page--404.html.twig
+++ b/templates/layout/page--404.html.twig
@@ -53,14 +53,16 @@
     <a href="#main-content" class="skiplink govuk-skip-link">Skip to main content</a>
   </div>
 </div>
+
 {% if govuk_cookie %}
   {{  govuk_cookie }}
 {% endif %}
-<header class="govuk-header " role="banner" data-module="govuk-header">
+
+<header class="govuk-header" id="global-header" role="banner" data-module="govuk-header">
   <div class="govuk-header__container govuk-width-container">
     {{ govuk_header }}
     {% if page.navigation %}
-      <div class="hamburger icon" onclick="toggleMobileMenu();">
+      <div id="mobile-menu-toggle" class="hamburger icon" onclick="toggleMobileMenu();" aria-expanded="false" aria-controls="mobile-menu">
         <i class="fa fa-bars"></i>
       </div>
       {{ page.navigation }}
@@ -69,8 +71,11 @@
       {{ search_box }}
     {% endif %}
   </div>
-  <div id="mobile-menu"></div>
+  {% if page.navigation %}
+    <div id="mobile-menu" aria-hidden="true"></div>
+  {% endif %}
 </header>
+
 <div class="govuk-width-container">
   {% if govuk_phase_banner %}
     {{  govuk_phase_banner }}
@@ -85,6 +90,7 @@
     <p>If the web address is correct or you selected a link or button, <a href="/contact">contact the {{ site_name}} {{site_slogan}} Helpline</a> if you need to speak to someone about this service.</p>
   </main>
 </div>
+
 <footer class="govuk-footer " role="contentinfo">
     <div class="govuk-width-container ">
       <div class="govuk-footer__meta">
diff --git a/templates/layout/page.html.twig b/templates/layout/page.html.twig
index 2f753fe..964c520 100755
--- a/templates/layout/page.html.twig
+++ b/templates/layout/page.html.twig
@@ -60,11 +60,11 @@
   </div>
 </div>
 
-<header class="govuk-header" role="banner" data-module="govuk-header">
+<header class="govuk-header" id="global-header" role="banner" data-module="govuk-header">
   <div class="govuk-header__container govuk-width-container">
     {{ govuk_header }}
     {% if page.navigation %}
-      <div class="hamburger icon" onclick="toggleMobileMenu();">
+      <div id="mobile-menu-toggle" class="hamburger icon" onclick="toggleMobileMenu();" aria-expanded="false" aria-controls="mobile-menu">
         <i class="fa fa-bars"></i>
       </div>
       {{ page.navigation }}
@@ -73,8 +73,11 @@
       {{ search_box }}
     {% endif %}
   </div>
-  <div id="mobile-menu"></div>
+  {% if page.navigation %}
+    <div id="mobile-menu" aria-hidden="true"></div>
+  {% endif %}
 </header>
+
 <div class="govuk-width-container">
   {% if govuk_phase_banner %}
     {{  govuk_phase_banner }}
