diff --git a/govuk.theme b/govuk.theme
index 3633671..9b351f4 100755
--- a/govuk.theme
+++ b/govuk.theme
@@ -421,6 +421,7 @@ function _govuk_theme_get_setting($name, $default = NULL) {
  * Implements hook_form_alter().
  */
 function govuk_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
+  $form['#attributes']['novalidate'] = 'novalidate';
   if (strpos($form_id, 'webform') !== FALSE) {
     $form['#attached']['library'][] = 'govuk/webform';
   }
@@ -433,3 +434,5 @@ function govuk_preprocess(&$variables) {
   $handler = \Drupal::service('theme_handler');
   $variables['govuk_directory'] = base_path() . $handler->getTheme('govuk')->getPath();
 }
+
+
diff --git a/templates/form/fieldset.html.twig b/templates/form/fieldset.html.twig
index 5b96567..0780985 100755
--- a/templates/form/fieldset.html.twig
+++ b/templates/form/fieldset.html.twig
@@ -7,6 +7,9 @@
     'govuk-fieldset',
   ]
 %}
+
+{% set error_id = attributes.id ~ '--status-message' %}
+
 <div class="govuk-form-group{% if errors %} govuk-form-group--error{% endif %}">
   <fieldset{{ attributes.addClass(classes) }}>
     {%
@@ -25,8 +28,8 @@
         <div{{ description.attributes.addClass('govuk-hint') }}>{{ description.content }}</div>
       {% endif %}
       {% if errors %}
-        <span class="govuk-error-message">
-          {{ errors }}
+        <span class="error-message govuk-error-message" id="{{ error_id }}">
+          {{ errors | striptags('<p>') }}
         </span>
       {% endif %}
       {% if prefix %}
diff --git a/templates/form/form-element.html.twig b/templates/form/form-element.html.twig
index ea37587..aadcf9a 100755
--- a/templates/form/form-element.html.twig
+++ b/templates/form/form-element.html.twig
@@ -71,6 +71,8 @@
   ]
 %}
 
+{% set error_id = 'edit-' ~ name|clean_class ~ '--status-message' %}
+
 {% if description_display == 'outside' and description.content and errors is empty %}
   <div{{ description.attributes.addClass(description_classes) }}>
       {{ description.content }}
@@ -83,7 +85,7 @@
   {% endif %}
 
   {% if errors and type not in ['radio', 'checkbox'] %}
-    <span class="error-message govuk-error-message">
+    <span class="error-message govuk-error-message" id="{{ error_id }}">
       {{ errors | striptags('<p>') }}
     </span>
   {% endif %}
diff --git a/templates/form/input--textfield.html.twig b/templates/form/input--textfield.html.twig
index 11dad12..cff7a63 100755
--- a/templates/form/input--textfield.html.twig
+++ b/templates/form/input--textfield.html.twig
@@ -19,5 +19,8 @@
   ]
 %}
 
+{% if attributes.placeholder is not empty %}
 <span class="form-hint govuk-hint">{{ attributes.placeholder }}</span>
+{% endif %}
+
 <input{{ attributes.addClass(classes).removeAttribute('placeholder') }} />{{ children }}
diff --git a/templates/layout/item-list.html.twig b/templates/layout/item-list.html.twig
index ff80ab7..e293753 100755
--- a/templates/layout/item-list.html.twig
+++ b/templates/layout/item-list.html.twig
@@ -19,7 +19,9 @@
  * @see template_preprocess_item_list()
  */
 #}
-{% if context.list_style %}
+{% if context.list_style == 'govuk-error-summary__list' %}
+{% set attributes = attributes.addClass('govuk-list govuk-error-summary__list') %}
+{% elseif context.list_style %}
   {%- set attributes = attributes.addClass('item-list__' ~ context.list_style) %}
 {% endif %}
 
