diff --git a/../password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php b/web/modules/contrib/password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php
index c296552b..6653b6c7 100644
--- a/../password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php
+++ b/web/modules/contrib/password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php
@@ -43,7 +43,7 @@ function validate($password, $user_context) {
     $strength = $password_strength->passwordStrength($password, $userData);
 
     if ($strength['score'] < $configuration['strength_score']) {
-      $validation->setErrorMessage($this->t('The password has a score of @password-score but the policy requires a score of at least @policy-score', array('@password-score'=>$strength['score'], '@policy-score'=>$configuration['strength_score'])));
+      $validation->setErrorMessage($this->t('The password was found in a list of commonly-used passwords. Please choose another.'));
     }
     return $validation;
   }
@@ -81,7 +81,7 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
    * {@inheritdoc}
    */
   public function getSummary() {
-    return $this->t('Password Strength minimum score of @score', array('@score' => $this->configuration['strength_score']));
+    return $this->t('Password must not be overly common.');
   }
 
 }
\ No newline at end of file
