Index: modules/contrib/file_entity/src/Controller/FileController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- modules/contrib/file_entity/src/Controller/FileController.php	(date 1503658397000)
+++ modules/contrib/file_entity/src/Controller/FileController.php	(revision )
@@ -12,6 +12,7 @@
 use Symfony\Component\HttpFoundation\BinaryFileResponse;
 use Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException;
 use Symfony\Component\HttpFoundation\Response;
+use Drupal\flysystem_s3\S3BinaryFileResponseObj;
 
 /**
  * Class FileController
@@ -74,13 +75,29 @@
     // Let other modules know the file is being downloaded.
     \Drupal::moduleHandler()->invokeAll('file_transfer', array($file->getFileUri(), $headers));
 
+    // Allow other modules to notify the FileController an s3 file is being transferred so a response object
+    // that supports s3 files maybe used instead of BinaryFileResponse.
+    $S3BasedFileObject = \Drupal::moduleHandler()->invokeAll('file_transfer_s3_storage', array($file->getFileUri()));
+
+    if ($S3BasedFileObject !== FALSE ) {
-    try {
+      try {
+        // Use a BinaryFileResponse that supports S3.
+        return new S3BinaryFileResponseObj($file->getFileUri(), 200, $headers);
+      }
+      catch (FileNotFoundException $e) {
+        return new Response(t('File @uri not found', array('@uri' =>$file->getFileUri())), 404);
+      }
+    }
+    else {
+      try {
+        // Default to local storage file behaviour.
-      return new BinaryFileResponse($file->getFileUri(), 200, $headers);
-    }
-    catch (FileNotFoundException $e) {
-      return new Response(t('File @uri not found', array('@uri' =>$file->getFileUri())), 404);
+        return new BinaryFileResponse($file->getFileUri(), 200, $headers);
+      }
+      catch (FileNotFoundException $e) {
+        return new Response(t('File @uri not found', array('@uri' =>$file->getFileUri())), 404);
+      }
     }
-  }
+    }
 
   /**
    * Return an Ajax dialog command for editing a file inline.
