diff --git a/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsDirectPartnership.feature b/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsDirectPartnership.feature
index 05a37db4c..2f271a867 100644
--- a/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsDirectPartnership.feature
+++ b/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsDirectPartnership.feature
@@ -7,25 +7,33 @@ Feature: Direct Partnership Happy Paths
     When the user visits the login page
     And the user logs in with the "par_authority@example.com" user credentials
     Then the user is on the dashboard page
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user creates a new "Direct" partnership application with the following details:
       | Authority | Partnership Info | AddressLine1  | AddressLine2 | Town | County     | Country        | Nation  | Postcode | Title | Firstname | Lastname | WorkNumber   | MobileNumber | Email                    |
       | Lower     | For Deletion     | 04 New Street | New Build    | Bury | Lancashire | United Kingdom | England | BL4 0BG  | Dr    | Steph     | Smith    | 017043356901 |  07704502913 | par_business@example.com |
     Then the first part of the partnership application is successfully completed
-    #second part of partnership application
+
+    # Second part of partnership application.
     When the user searches for the last created partnership
     And the user completes the partnership application with the following details:
       | Business Description | ContactNotes | SIC Code            | No of Employees | Trading Name | Legal Entity Name | Legal entity Type | Company number |
       | Test Business        | Test Note.   | allow people to eat | 10 to 49        | Testing Co.  | Testing Co.       | Partnership       |       12345678 |
     Then the second part of the partnership application is successfully completed
     And the user signs out
-    # Nominate the Partnership
+
+    # Nominate the Partnership.
     Given the user is on the PAR login page
     And the user logs in with the "secretary_state@example.com" user credentials
     When the user searches for the last created partnership
     And the user approves the partnership
     And the user searches again for the last created partnership
     Then the partnership is displayed with Status "Active" and Actions "Revoke partnership"
-    # Delete the Partnership
+
+    # Delete the Partnership.
     When the user Deletes the Partnership with the following reason: "Partnership is incorrect."
     Then the Partnership was Deleted Successfully
     And the user signs out
@@ -75,6 +83,11 @@ Feature: Direct Partnership Happy Paths
   Scenario: Update all Partnership details successfully (Happy Path - PAR-2214)
     Given the user is on the PAR login page
     And the user logs in with the "par_helpdesk@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the last created partnership Authority
     And the user updates the About the Partnership and Regulatory Functions:
       | About the Partnership |
@@ -129,6 +142,11 @@ Feature: Direct Partnership Happy Paths
   Scenario: Verify the Removal of a Legal Entity from an Active Partnership (Happy Path - PAR-2314)
     Given the user is on the PAR login page
     And the user logs in with the "senior_administrator@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the last created partnership Authority
     And the user removes the legal entity
     Then the user verifies the legal entity was removed successfully
@@ -233,13 +251,19 @@ Feature: Direct Partnership Happy Paths
   Scenario: Verify the Sending and Approval of a Notification of Proposed Enforcement (Happy Path - PAR-1852, PAR-1853, PAR-1854)
     Given the user is on the PAR login page
     And the user logs in with the "par_enforcement_officer@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the last created partnership
     And the user creates an enforcement notice against the partnership with the following details:
       | Enforcement Action | Title               | Regulatory Function | Description                  | Attachment |
       | Proposed           | Enforcement Title 1 | Alphabet learning   | Test Enforcement Description | link.txt   |
     Then all the fields for the enforcement notice are updated correctly
     And the user signs out
-    #	Approve the Enforcement Notice
+
+    # Approve the Enforcement Notice.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
     When the user selects the last created enforcement notice
@@ -251,6 +275,11 @@ Feature: Direct Partnership Happy Paths
   Scenario: Verify a Help Desk user can remove an Enforcement Notice Successfully (Happy Path - PAR-1855)
     Given the user is on the PAR login page
     And the user logs in with the "secretary_state@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the last created enforcement notice
     And the user removes the enforcement notice
     Then the enforcement notice is removed successfully
@@ -266,7 +295,8 @@ Feature: Direct Partnership Happy Paths
       | Proposed           | Enforcement Notice to be Blocked | Alphabet learning   | Test Enforcement. | link.txt   |
     Then all the fields for the enforcement notice are updated correctly
     And the user signs out
-    # Block the Enforcement Notice
+
+    # Block the Enforcement Notice.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
     When the user selects the last created enforcement notice
@@ -284,9 +314,15 @@ Feature: Direct Partnership Happy Paths
       | Test Enforcement Notice Discussion. |
     Then the Enquiry is created Successfully
     And the user signs out
-    # Login as the Authority and Verify the General Enquiry
+
+    # Login as the Authority and Verify the General Enquiry.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the last created general enquiry
     Then the general enquiry is recieved successfully
     And the user signs out
@@ -301,7 +337,8 @@ Feature: Direct Partnership Happy Paths
       | Test Deviation Request to be Blocked. |
     Then the Deviation Request is created Successfully
     And the user signs out
-    # Login as the Authority and Verify the Deviation Request
+
+    # Login as the Authority and Verify the Deviation Request.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
     When the user searches for the last created deviation request
@@ -319,7 +356,8 @@ Feature: Direct Partnership Happy Paths
       | Test Deviation Request. |
     Then the Deviation Request is created Successfully
     And the user signs out
-    # Login as primary authority and check and approve deviation request
+
+    # Login as primary authority and check and approve deviation request.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
     When the user searches for the last created deviation request
@@ -330,7 +368,8 @@ Feature: Direct Partnership Happy Paths
       | Authority Response. |
     Then the response is displayed successfully
     And the user signs out
-    # Submit a Response as the Enforcement Officer
+
+    # Submit a Response as the Enforcement Officer.
     Given the user is on the PAR login page
     And the user logs in with the "par_enforcement_officer@example.com" user credentials
     When the user searches for the last created deviation request
@@ -339,7 +378,8 @@ Feature: Direct Partnership Happy Paths
       | Enforcement Officer Reply. |
     Then the response is displayed successfully
     And the user signs out
-    # Submit a Response as the Help Desk
+
+    # Submit a Response as the Help Desk.
     Given the user is on the PAR login page
     And the user logs in with the "par_helpdesk@example.com" user credentials
     When the user searches for the last created deviation request
diff --git a/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsOther.feature b/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsOther.feature
index 9bd75cce6..871daa8a3 100644
--- a/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsOther.feature
+++ b/tests/e2e/par-test-automation/src/test/resources/features/HappyPathsOther.feature
@@ -26,7 +26,7 @@ Feature: Other Happy Paths
       | Authority Type | ONS Code | Regulatory Function | AddressLine1  | AddressLine2 | Town    | County         | Country        | Nation  | Postcode |
       | Council Area   | 43453465 | Cookie control      | 32 Bramtom Rd | new build    | Windsor | Greater London | United Kingdom | England | SL4 5PN  |
     Then the authority is created sucessfully
-    #Update All Fields for newly created Authority
+    # Update All Fields for newly created Authority.
     When the user searches for the last created authority
     And the user updates all the fields for newly created authority
     Then the update for the authority is successful
@@ -54,10 +54,12 @@ Feature: Other Happy Paths
   Scenario: Verify an Authority contact is subscribed to PAR News (Happy Path - PAR-2076)
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
+
     When the user selects a contact to update
     Then the user can successfully subscribe to PAR News
     And the user signs out
-    #Helpdesk Verification
+
+    # Helpdesk Verification
     Given the user is on the PAR login page
     And the user logs in with the "par_helpdesk@example.com" user credentials
     When the user is on the Subscriptions page
@@ -115,16 +117,24 @@ Feature: Other Happy Paths
     When the user searches for the "par_authority_manager@example.com" user account
     And the user clicks the manage contact link
     Then the user can view the user account successfully
+
     # Change the user role.
     When the user changes the users role to "Authority Manager"
     Then the user role was changed successfully
     And the user signs out
+
     # Verify the Authority Manager can change the Authority Member's role.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority_manager@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user searches for the "par_authority@example.com" user account
     And the user clicks the manage contact link
     Then the user can view the user account successfully
+
     # Change the user role.
     When the user changes the users role to "Authority Manager"
     Then the user role was changed successfully
@@ -179,7 +189,7 @@ Feature: Other Happy Paths
     Then the user is on the dashboard page
     And the user signs out
 
-  @regression @happypath @helpDesk @usermanagement
+  @regression @happypath @helpDesk @usermanagement @par_2603
   Scenario: Verify the Addition and Update of a New Persons Contact Record as a Help Desk User (Happy Path - PAR-2097)
     Given the user is on the PAR login page
     And the user logs in with the "par_helpdesk@example.com" user credentials
@@ -195,7 +205,7 @@ Feature: Other Happy Paths
     Then the user can verify the person was updated successfully and can send an account invitation
     And the user signs out
 
-  @regression @happypath @helpDesk @usermanagement
+  @regression @happypath @helpDesk @usermanagement @par_2603
   Scenario: Verify a New User can be invited to create a User Account (Happy Path - PAR-2457)
     Given the user is on the PAR login page
     And the user logs in with the "par_helpdesk@example.com" user credentials
@@ -207,8 +217,8 @@ Feature: Other Happy Paths
     Then the user verifies the user account invitation was sent successfully
     And the user signs out
 
-  @regression @happypath @usermanagement
-  Scenario: Verify Completion of User Creation journey (Happy Path - PAR-1904)
+  @regression @happypath @usermanagement @par_1904 @par_2603
+  Scenario: Verify Completion of User Creation journey (Happy Path - PAR-1904 PAR-2603)
     Given the user is on the PAR login page
     And the user logs in with the "par_admin@example.com" user credentials
     When the user visits the maillog page and extracts the invite link
@@ -222,10 +232,16 @@ Feature: Other Happy Paths
   Scenario: Verify a User can Change their User Account Email Address (Happy Path - PAR-2323)
     Given the user is on the PAR login page
     And the user logs in with the "par_authority@example.com" user credentials
+
+    # Hide the cookies banner so that more is displayed in the screen shots.
+    When the user accepts the analytics cookies
+    Then analytical cookies have been accepted successfully
+
     When the user updates their user account email address to "par_authority_2@example.com"
     Then the user can verify the new email address is displayed on the header
     And the user signs out
-    # Log in with the new Email Address and revertback to the original Email Address
+
+    # Log in with the new Email Address and revert back to the original Email Address.
     Given the user is on the PAR login page
     And the user logs in with the "par_authority_2@example.com" user credentials
     When the user updates their user account email address to "par_authority@example.com"
