diff --git a/web/sites/default/settings.php b/web/sites/default/settings.php
--- a/web/sites/default/settings.php	(revision 71847c0ceff38efeb0143cb38d4a0ef070c1498d)
+++ b/web/sites/default/settings.php	(date 1745926499926)
@@ -908,7 +908,9 @@
   $settings['redis.connection']['scheme'] = $redis_credentials->scheme ?? 'tls';
   $settings['redis.connection']['host'] = $redis_credentials->host;
   $settings['redis.connection']['port'] = $redis_credentials->port;
-  $settings['redis.connection']['password'] = $redis_credentials->password;
+  if ($redis_credentials->password) {
+    $settings['redis.connection']['password'] = $redis_credentials->password;
+  }
   $settings['cache']['default'] = 'cache.backend.redis';
   $settings['cache']['bins']['form'] = 'cache.backend.database';
   $settings['cache']['bins']['render'] = 'cache.backend.database';
@@ -961,12 +963,13 @@
   ];
 }

-// Set the PaaS opensearch conneciton credentials.
+// Set the PaaS opensearch connection credentials.
 if (isset($os_credentials)) {
   $config['search_api.server.opensearch']['backend_config']['connector'] = 'basicauth';
   $config['search_api.server.opensearch']['backend_config']['connector_config']['url'] = $os_credentials->uri;
   $config['search_api.server.opensearch']['backend_config']['connector_config']['username'] = $os_credentials->username;
   $config['search_api.server.opensearch']['backend_config']['connector_config']['password'] = $os_credentials->password;
+  $config['search_api.server.opensearch']['backend_config']['connector_config']['ssl_verification'] = $os_credentials->ssl_verification ?? FALSE;
 }

 // Set flysystem configuration to use local files for all environments,
