diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/AdvicePageObjects/AdviceNoticeSearchPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/AdvicePageObjects/AdviceNoticeSearchPage.java
index 936bf2675..cf3350575 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/AdvicePageObjects/AdviceNoticeSearchPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/AdvicePageObjects/AdviceNoticeSearchPage.java
@@ -33,6 +33,7 @@ public class AdviceNoticeSearchPage extends BasePageObject {
 		adviceSearchBar.sendKeys(title);

 		searchBtn.click();
+		waitForPageLoad();
 	}

 	public void selectUploadLink() {
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/BasePageObject.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/BasePageObject.java
index 900ccaac6..35a4a71b3 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/BasePageObject.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/BasePageObject.java
@@ -1,6 +1,7 @@
 package uk.gov.beis.pageobjects;

 import java.io.BufferedReader;
+import java.io.File;
 import java.io.FileReader;
 import java.io.IOException;
 import java.util.Map;
@@ -76,8 +77,10 @@ public class BasePageObject {
 	}

 	public void uploadDocument(WebElement filebrowser, String filename) {
+	    // For Docksal.
+	    String workingDirectory = System.getProperty("app.dir").toString();
 		if(BrowserFactory.browser == Browser.Chrome) {
-			filebrowser.sendKeys(System.getProperty("user.dir") + (System.getProperty("user.dir").toString().endsWith("/") ? "" : "/") + filename);
+			filebrowser.sendKeys(workingDirectory + (workingDirectory.endsWith(File.separator) ? "" : File.separator) + filename);
 		}
 		else if (BrowserFactory.browser == Browser.Firefox) {
 			filebrowser.sendKeys(System.getProperty("user.dir") + "\\" + filename);
@@ -91,8 +94,10 @@ public class BasePageObject {
 	public String[] getColumnFromCSV(String filename, int column, String delimiter) {
 		String path = "";

+	    // For Docksal.
+	    String workingDirectory = System.getProperty("app.dir").toString();
 		if(BrowserFactory.browser == Browser.Chrome) {
-			path = System.getProperty("user.dir") + (System.getProperty("user.dir").toString().endsWith("/") ? "" : "/") + filename;
+            path = workingDirectory + (workingDirectory.endsWith(File.separator) ? "" : File.separator) + filename;
 		}
 		else if (BrowserFactory.browser == Browser.Firefox) {
 			path = System.getProperty("user.dir") + "\\" + filename;
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/EnforcementNoticePageObjects/EnforcementActionPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/EnforcementNoticePageObjects/EnforcementActionPage.java
index 474d7b543..1c45111ce 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/EnforcementNoticePageObjects/EnforcementActionPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/EnforcementNoticePageObjects/EnforcementActionPage.java
@@ -60,7 +60,7 @@ public class EnforcementActionPage extends BasePageObject {
 	public void chooseFile(String filename) {
 		uploadDocument(chooseFile, filename);

-		DataStore.saveValue(UsableValues.ENFORCEMENT_FILENAME, filename.replace(".txt", ""));
+		DataStore.saveValue(UsableValues.ENFORCEMENT_FILENAME, filename.replace(".txt", "").substring(1));
 	}

 	public void clearAllFields() {
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/GeneralEnquiryPageObjects/EnquiryReviewPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/GeneralEnquiryPageObjects/EnquiryReviewPage.java
index 2a3062234..e0a8587f9 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/GeneralEnquiryPageObjects/EnquiryReviewPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/GeneralEnquiryPageObjects/EnquiryReviewPage.java
@@ -34,13 +34,13 @@ public class EnquiryReviewPage extends BasePageObject {

 	public boolean checkEnquiryDetails() {
 		WebElement enquiryDescription = driver.findElement(By.xpath(descriptionLocator.replace("?", DataStore.getSavedValue(UsableValues.ENQUIRY_DESCRIPTION))));
-		WebElement file = driver.findElement(By.xpath(fileLocator.replace("?", "link")));
+		WebElement file = driver.findElement(By.xpath(fileLocator.replace("?", "ink")));
 		return enquiryDescription.isDisplayed() && file.isDisplayed();
 	}

 	public boolean checkEnquiryResponse() {
 		WebElement reply = driver.findElement(By.xpath(responseLocator.replace("?", DataStore.getSavedValue(UsableValues.MESSAGE_RESPONSE))));
-		WebElement file = driver.findElement(By.xpath(fileLocator.replace("?", "link")));
+		WebElement file = driver.findElement(By.xpath(fileLocator.replace("?", "ink")));
 		return reply.isDisplayed() && file.isDisplayed();
 	}

diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/InspectionPlanFeedbackPageObjects/InspectionFeedbackConfirmationPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/InspectionPlanFeedbackPageObjects/InspectionFeedbackConfirmationPage.java
index 2d0a04628..a78628827 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/InspectionPlanFeedbackPageObjects/InspectionFeedbackConfirmationPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/InspectionPlanFeedbackPageObjects/InspectionFeedbackConfirmationPage.java
@@ -34,7 +34,7 @@ public class InspectionFeedbackConfirmationPage extends BasePageObject {

 	public boolean checkInspectionResponse() {
 		WebElement response = driver.findElement(By.xpath(responseLocator.replace("?", DataStore.getSavedValue(UsableValues.MESSAGE_RESPONSE))));
-		WebElement fileUpload = driver.findElement(By.xpath(fileLocator.replace("?", "link")));
+		WebElement fileUpload = driver.findElement(By.xpath(fileLocator.replace("?", "ink")));
 		return response.isDisplayed() && fileUpload.isDisplayed();
 	}

diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/LegalEntityPageObjects/ConfirmThisAmendmentPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/LegalEntityPageObjects/ConfirmThisAmendmentPage.java
index cec17d345..589ebf0d2 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/LegalEntityPageObjects/ConfirmThisAmendmentPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/LegalEntityPageObjects/ConfirmThisAmendmentPage.java
@@ -9,33 +9,34 @@ import org.openqa.selenium.support.FindBy;
 import uk.gov.beis.pageobjects.BasePageObject;

 public class ConfirmThisAmendmentPage extends BasePageObject {
-
+
 	@FindBy(id = "edit-confirmation")
 	private WebElement confirmationCheckbox;
-
+
 	@FindBy(id = "edit-next")
 	private WebElement submitAmendmentBtn;
-
+
 	public ConfirmThisAmendmentPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void deselectConfirmationCheckbox() {
 		if(confirmationCheckbox.isSelected()) {
 			confirmationCheckbox.click();
 		}
 	}
-
+
 	public void selectConfirmationCheckbox() {
 		if(!confirmationCheckbox.isSelected()) {
 			confirmationCheckbox.click();
 		}
 	}
-
+
 	public void selectSubmitAmendmentButton() {
 		submitAmendmentBtn.click();
+		waitForPageLoad();
 	}
-
+
 	public void goToAmendmentCompletedPage() {
 		WebElement element = driver.findElement(By.xpath("//input[@value='Confirm amendment']"));
 		executeJavaScript("arguments[0].click();", element);
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/OtherPageObjects/LoginPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/OtherPageObjects/LoginPage.java
index c79fbe9df..6338bbe5d 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/OtherPageObjects/LoginPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/OtherPageObjects/LoginPage.java
@@ -31,8 +31,7 @@ public class LoginPage extends BasePageObject {
 	}

 	public void navigateToUrl() {
-		//ScenarioContext.lastDriver.get(PropertiesUtil.getConfigPropertyValue("par_url") + "/user/login?current");
-        ScenarioContext.lastDriver.get(PropertiesUtil.getConfigPropertyValue("par_url") + "/user/login%3Fcurrent");
+        ScenarioContext.lastDriver.get(PropertiesUtil.getConfigPropertyValue("par_url") + "/user/login");
 		checkLoginPage();
 	}

diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/AuthorityPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/AuthorityPage.java
index 41859faad..0c0dd8a83 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/AuthorityPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/AuthorityPage.java
@@ -16,20 +16,22 @@ public class AuthorityPage extends BasePageObject {
 	private WebElement continueBtn;

 	private String authorityLocator = "//label[contains(text(),'?')]";
-
+
 	public AuthorityPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void selectAuthority(String auth) {
 		WebElement authority = driver.findElement(By.xpath(authorityLocator.replace("?", auth)));
-
+
 		DataStore.saveValue(UsableValues.AUTHORITY_NAME, authority.getText());
-
+
 		authority.click();
 	}
-
+
 	public void selectContinueButton() {
-		continueBtn.click();
+        waitForElementToBeClickable(By.id("edit-next"), 3000);
+        continueBtn.click();
+        waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/BusinessNamePage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/BusinessNamePage.java
index 67a347734..d92443224 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/BusinessNamePage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/BusinessNamePage.java
@@ -8,29 +8,30 @@ import org.openqa.selenium.support.FindBy;
 import uk.gov.beis.pageobjects.BasePageObject;

 public class BusinessNamePage extends BasePageObject {
-
+
 	@FindBy(id = "edit-name")
 	private WebElement businessName;
-
+
 	@FindBy(id = "edit-next")
 	private WebElement continueBtn;
-
+
 	@FindBy(id = "edit-save")
 	private WebElement saveBtn;
-
+
 	public BusinessNamePage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void enterBusinessName(String name) {
 		businessName.clear();
 		businessName.sendKeys(name);
 	}
-
+
 	public void clickContinueButton() {
 		continueBtn.click();
+		waitForPageLoad();
 	}
-
+
 	public void clickSaveButton() {
 		saveBtn.click();
 	}
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/MemberListCountPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/MemberListCountPage.java
index 897e37e02..1a878b4bf 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/MemberListCountPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/MemberListCountPage.java
@@ -29,5 +29,6 @@ public class MemberListCountPage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeVisible(By.id("edit-next"), 2000);
 		continueBtn.click();
+		waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTermsPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTermsPage.java
index 83c02af88..5d1509233 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTermsPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTermsPage.java
@@ -39,5 +39,6 @@ public class PartnershipTermsPage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeClickable(By.id("edit-next"), 2000);
         continueBtn.click();
+        waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTypePage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTypePage.java
index 063cf6f96..8cc9c4617 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTypePage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/PartnershipTypePage.java
@@ -26,5 +26,6 @@ public class PartnershipTypePage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeClickable(By.id("edit-next"), 3000);
 		continueBtn.click();
+		waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/RegulatoryFunctionPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/RegulatoryFunctionPage.java
index 2c6790d5b..12f41e38e 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/RegulatoryFunctionPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/PartnershipPageObjects/RegulatoryFunctionPage.java
@@ -95,5 +95,6 @@ public class RegulatoryFunctionPage extends BasePageObject {
 	public void selectSaveButton() {
         waitForElementToBeClickable(By.id("edit-save"), 2000);
         saveBtn.click();
+        waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/AddAddressPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/AddAddressPage.java
index 85780b548..e3e243851 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/AddAddressPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/AddAddressPage.java
@@ -95,6 +95,7 @@ public class AddAddressPage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeClickable(By.id("edit-next"), 2000);
 		continueBtn.click();
+		waitForPageLoad();
 	}

 	public void clickSaveButton() {
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/DeclarationPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/DeclarationPage.java
index 53b6f119e..d9ddc045a 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/DeclarationPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/SharedPageObjects/DeclarationPage.java
@@ -63,6 +63,7 @@ public class DeclarationPage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeClickable(By.id("edit-next"), 2000);
         continueBtn.click();
+        waitForPageLoad();
 	}

 	public void goToEnforcementSearchPage() {
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/TransferPartnerships/PartnershipMigrationSelectionPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/TransferPartnerships/PartnershipMigrationSelectionPage.java
index 79f7ebd9e..2d8201448 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/TransferPartnerships/PartnershipMigrationSelectionPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/TransferPartnerships/PartnershipMigrationSelectionPage.java
@@ -36,5 +36,6 @@ public class PartnershipMigrationSelectionPage extends BasePageObject {

         waitForElementToBeVisible(By.id("edit-next"), 2000);
         continueBtn.click();
+        waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/BaseDashboardPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/BaseDashboardPage.java
index 4f97acf4e..cd9dc2d90 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/BaseDashboardPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/BaseDashboardPage.java
@@ -38,7 +38,7 @@ public class BaseDashboardPage extends BasePageObject{
 	}

 	public void hideCookieBanner() {
-		//cookieBanner.findElement(By.xpath("//button[contains(text(),'Hide this message')]")).click();
+		cookieBanner.findElement(By.xpath("//button[contains(text(),'Hide this message')]")).click();
 	}

 	//public Boolean checkCookiesAccepted() {
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/HelpDeskDashboardPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/HelpDeskDashboardPage.java
index 05a97c141..3f5ad9ede 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/HelpDeskDashboardPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserDashboardPageObjects/HelpDeskDashboardPage.java
@@ -6,80 +6,81 @@ import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;

 public class HelpDeskDashboardPage extends BaseDashboardPage {
-
+
 	@FindBy(linkText  = "View all statistics")
 	private WebElement viewStatisticsLink;
-
+
 	@FindBy(linkText = "Search partnerships")
 	private WebElement searchPartnershipBtn;
-
+
 	@FindBy(linkText = "Manage authorities")
 	private WebElement mangeAuthoritiesBtn;
-
+
 	@FindBy(linkText = "Manage organisations")
 	private WebElement mangeOrganisationsBtn;
-
+
 	@FindBy(linkText = "Manage people")
 	private WebElement managePeopleBtn;
-
+
 	@FindBy(linkText = "Manage subscriptions")
 	private WebElement manageSubscriptionsBtn;
-
+
 	@FindBy(linkText = "Manage enforcement notices")
 	private WebElement manageEnforcementNoticesBtn;
-
+
 	@FindBy(linkText = "Manage general enquiries")
 	private WebElement manageGeneralEnquiriesBtn;
-
+
 	@FindBy(linkText = "Manage deviation requests")
 	private WebElement manageDeviationRequestLink;
-
+
 	@FindBy(linkText = "Manage inspection feedback")
 	private WebElement manageInspectionFeedbackLink;
-
+
 	@FindBy(linkText = "Manage general enquiries")
 	private WebElement manageGeneralEnquiryLink;
-
+
 	public HelpDeskDashboardPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void selectViewAllStatistics() {
 		viewStatisticsLink.click();
 	}
-
+
 	public void selectSearchPartnerships() {
 		searchPartnershipBtn.click();
 	}
-
+
 	public void selectManageAuthorities() {
 		mangeAuthoritiesBtn.click();
 	}
-
+
 	public void selectManageOrganisations() {
 		mangeOrganisationsBtn.click();
 	}
-
+
 	public void selectManagePeople() {
 		managePeopleBtn.click();
+		waitForPageLoad();
 	}
-
+
 	public void selectManageSubscriptions() {
 		manageSubscriptionsBtn.click();
 	}
-
+
 	public void selectManageEnforcementNotices() {
 		manageEnforcementNoticesBtn.click();
 	}
-
+
 	public void selectManageDeviationRequests() {
 		manageDeviationRequestLink.click();
 	}
-
+
 	public void selectManageInspectionFeedback() {
 		manageInspectionFeedbackLink.click();
 	}
-
+
 	public void selectManageGeneralEnquiry() {
 		manageGeneralEnquiryLink.click();
 	}
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/AddMembershipConfirmationPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/AddMembershipConfirmationPage.java
index a0b455193..df1009f41 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/AddMembershipConfirmationPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/AddMembershipConfirmationPage.java
@@ -20,5 +20,6 @@ public class AddMembershipConfirmationPage extends BasePageObject {
 	public void clickContinueButton() {
         waitForElementToBeVisible(By.id("edit-next"), 2000);
 		continueBtn.click();
+		waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ChoosePersonToAddPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ChoosePersonToAddPage.java
index 434d08bd0..63b1fe38c 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ChoosePersonToAddPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ChoosePersonToAddPage.java
@@ -11,25 +11,26 @@ import uk.gov.beis.pageobjects.BasePageObject;
 import uk.gov.beis.utility.DataStore;

 public class ChoosePersonToAddPage extends BasePageObject{
-
+
 	@FindBy(id = "edit-next")
 	private WebElement continueBtn;
-
+
 	private String personsNameLocator = "//label/div[contains(normalize-space(), '?')]/../../input";
-
+
 	public ChoosePersonToAddPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void choosePerson() {
 		WebElement radio = driver.findElement(By.xpath(personsNameLocator.replace("?", DataStore.getSavedValue(UsableValues.PERSON_FULLNAME_TITLE))));
-
+
 		if(radio != null) {
 			radio.click();
 		}
 	}
-
+
 	public void clickContinueButton() {
 		continueBtn.click();
+		waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ContactRecordsPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ContactRecordsPage.java
index 6cca8c25e..05b2d43d2 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ContactRecordsPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ContactRecordsPage.java
@@ -9,24 +9,26 @@ import org.openqa.selenium.support.FindBy;
 import uk.gov.beis.pageobjects.BasePageObject;

 public class ContactRecordsPage extends BasePageObject {
-
+
 	@FindBy(name = "user_person")
 	private List<WebElement> userContacts;
-
+
 	@FindBy(id = "edit-next")
 	private WebElement continueBtn;
-
+
 	public ContactRecordsPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void selectContactToUpdate() {
 		if(userContacts.size() > 0) {
 			userContacts.get(0).click();
+			waitForPageLoad();
 		}
 	}
-
+
 	public void selectContinueButton() {
 		continueBtn.click();
-	}
+	   	waitForPageLoad();
+    }
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/GiveUserAccountPage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/GiveUserAccountPage.java
index aa5fcea85..107fa602f 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/GiveUserAccountPage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/GiveUserAccountPage.java
@@ -8,29 +8,30 @@ import org.openqa.selenium.support.FindBy;
 import uk.gov.beis.pageobjects.BasePageObject;

 public class GiveUserAccountPage extends BasePageObject {
-
+
 	@FindBy(id = "edit-account-new")
 	private WebElement inviteUserRadioBtn;
-
+
 	@FindBy(xpath = "//label[contains(text(),'Use the existing account:')]")
 	private WebElement existingAccountRadial;
-
+
 	@FindBy(id = "edit-next")
 	private WebElement continueBtn;
-
+
 	public GiveUserAccountPage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void selectInviteUserToCreateAccount() {
 		inviteUserRadioBtn.click();
 	}
-
+
 	public void selectUseExistingAccount() {
 		existingAccountRadial.click();
 	}
-
+
 	public void clickContinueButton() {
 		continueBtn.click();
+        waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ManagePeoplePage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ManagePeoplePage.java
index 25b284a61..661f1ee93 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ManagePeoplePage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/ManagePeoplePage.java
@@ -11,66 +11,67 @@ import uk.gov.beis.pageobjects.BasePageObject;
 import uk.gov.beis.utility.DataStore;

 public class ManagePeoplePage extends BasePageObject {
-
+
 	@FindBy(linkText = "Add a person")
 	private WebElement addPersonBtn;
-
+
 	@FindBy(id = "edit-name-email-search")
 	private WebElement searchTextField;
-
+
 	@FindBy(id = "edit-authority-organisation-filter")
 	private WebElement authorityOrganisationTextField;
-
+
 	@FindBy(id = "edit-submit-par-people")
 	private WebElement submitBtn;
-
+
 	@FindBy(xpath = "//td[@class='views-field views-field-last-name']")
 	private WebElement personNameTableElement;
-
+
 	@FindBy(linkText = "Manage contact")
 	private WebElement manageContactBtn;
-
+
 	private String getNameColumnLocator = "//td[contains(normalize-space(), '?')]/preceding-sibling::td";
 	private String manageContactByEmailLocator = "//td[contains(normalize-space(), '?')]/following-sibling::td[2]/a[contains(normalize-space(), 'Manage contact')]";
 	private String manageContactByNameLocator = "//td[contains(normalize-space(), '?')]/following-sibling::td[3]/a[contains(normalize-space(), 'Manage contact')]";
-
+
 	public ManagePeoplePage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void selectAddPerson() {
 		addPersonBtn.click();
 	}
-
+
 	public void enterNameOrEmail(String searchText) {
 		searchTextField.sendKeys(searchText);
 	}
-
+
 	public void enterAuthorityOrOrganisation(String searchText) {
 		authorityOrganisationTextField.sendKeys(searchText);
 	}
-
+
 	public void clickSubmit() {
 		submitBtn.click();
+		waitForPageLoad();
 	}
-
+
 	public String GetPersonName() {
 		WebElement nameColumn = driver.findElement(By.xpath(getNameColumnLocator.replace("?", DataStore.getSavedValue(UsableValues.PERSON_EMAIL_ADDRESS).toLowerCase())));
 		return nameColumn.getText().trim();
-
+
 		//return personNameTableElement.getText().trim();
 	}
-
+
 	public void findManageContactByEmail(String personEmail) {
 		WebElement link = driver.findElement(By.xpath(manageContactByEmailLocator.replace("?", personEmail)));
 		link.click();
 	}
-
+
 	public void findManageContactByName(String personName) {
 		WebElement link = driver.findElement(By.xpath(manageContactByNameLocator.replace("?", personName)));
 		link.click();
 	}
-
+
 	public void clickManageContact() {
 		manageContactBtn.click();
 	}
diff --git a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/UserRoleTypePage.java b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/UserRoleTypePage.java
index a9ef1e628..db9ad5900 100644
--- a/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/UserRoleTypePage.java
+++ b/tests/e2e/par-test-automation/src/main/java/uk/gov/beis/pageobjects/UserManagement/UserRoleTypePage.java
@@ -10,51 +10,51 @@ import org.openqa.selenium.support.FindBy;
 import uk.gov.beis.pageobjects.BasePageObject;

 public class UserRoleTypePage extends BasePageObject {
-
+
 	@FindBy(id = "edit-general-secretary-state")
 	private WebElement secretaryOfStateBtn;
-
+
 	@FindBy(id = "edit-general-senior-administration-officer")
 	private WebElement seniorAdministratorBtn;
-
+
 	@FindBy(id = "edit-general-par-helpdesk")
 	private WebElement processingTeamMemberBtn;
-
+
 	@FindBy(id = "edit-general-national-regulator")
 	private WebElement nationalRegulatorBtn;
-
+
 	@FindBy(id = "edit-par-data-authority-par-authority-manager")
 	private WebElement authorityManagerBtn;
-
+
 	@FindBy(id = "edit-par-data-authority-par-authority")
 	private WebElement authorityMemberBtn;
-
+
 	@FindBy(id = "edit-par-data-authority-par-enforcement")
 	private WebElement enforcementOfficerBtn;
-
+
 	@FindBy(id = "edit-par-data-organisation-par-organisation-manager")
 	private WebElement organisationManagerBtn;
-
+
 	@FindBy(id = "edit-par-data-organisation-par-organisation")
 	private WebElement organisationMemberBtn;
-
+
 	@FindBy(id = "edit-next")
 	private WebElement continueBtn;
-
+
 	public UserRoleTypePage() throws ClassNotFoundException, IOException {
 		super();
 	}
-
+
 	public void deselectAllMemberships() {
 		List<WebElement> checkboxes = driver.findElements(By.xpath("//input[@class='form-checkbox form-control govuk-input govuk-checkboxes__input']"));
-
+
 		for(WebElement checkbox : checkboxes) {
 			if(checkbox.isSelected()) {
 				checkbox.click();
 			}
 		}
 	}
-
+
 	public void chooseMembershipRole(String roleType) {
 		switch(roleType) {
 		case "Secretary of State":
@@ -86,7 +86,7 @@ public class UserRoleTypePage extends BasePageObject {
 			break;
 		}
 	}
-
+
 	public void selectSecretaryOfState() {
 		if(!secretaryOfStateBtn.isSelected()) {
 			secretaryOfStateBtn.click();
@@ -97,50 +97,51 @@ public class UserRoleTypePage extends BasePageObject {
 			seniorAdministratorBtn.click();
 		}
 	}
-
+
 	public void selectProcessingTeamMember() {
 		if(!processingTeamMemberBtn.isSelected()) {
 			processingTeamMemberBtn.click();
 		}
 	}
-
+
 	public void selectNationalRegulator() {
 		if(!nationalRegulatorBtn.isSelected()) {
 			nationalRegulatorBtn.click();
 		}
 	}
-
+
 	public void selectAuthorityManager() {
 		if(!authorityManagerBtn.isSelected()) {
 			authorityManagerBtn.click();
 		}
 	}
-
+
 	public void selectAuthorityMember() {
 		if(!authorityMemberBtn.isSelected()) {
 			authorityMemberBtn.click();
 		}
 	}
-
+
 	public void selectEnforcementOfficer() {
 		if(!enforcementOfficerBtn.isSelected()) {
 			enforcementOfficerBtn.click();
 		}
 	}
-
+
 	public void selectOrganisationManager() {
 		if(!organisationManagerBtn.isSelected()) {
 			organisationManagerBtn.click();
 		}
 	}
-
+
 	public void selectOrganisationMember() {
 		if(!organisationMemberBtn.isSelected()) {
 			organisationMemberBtn.click();
 		}
 	}
-
+
 	public void clickContinueButton() {
 		continueBtn.click();
+		waitForPageLoad();
 	}
 }
diff --git a/tests/e2e/par-test-automation/src/test/java/uk/gov/beis/test/PARTestRunner.java b/tests/e2e/par-test-automation/src/test/java/uk/gov/beis/test/PARTestRunner.java
index 105f2f230..b38d05af6 100644
--- a/tests/e2e/par-test-automation/src/test/java/uk/gov/beis/test/PARTestRunner.java
+++ b/tests/e2e/par-test-automation/src/test/java/uk/gov/beis/test/PARTestRunner.java
@@ -7,7 +7,7 @@ import io.cucumber.junit.Cucumber;
 import io.cucumber.junit.CucumberOptions;

 @RunWith(Cucumber.class)
-@CucumberOptions(dryRun = false, features = { "classpath:features" }, glue = { "uk.gov.beis.stepdefs" }, tags = "@sadpath", plugin = { "json:target/cucumber-report/report.json", "html:target/cucumber-report/cucumber.html" })
+@CucumberOptions(dryRun = false, features = { "classpath:features" }, glue = { "uk.gov.beis.stepdefs" }, tags = "@regression", plugin = { "json:target/cucumber-report/report.json", "html:target/cucumber-report/cucumber.html" })
 public class PARTestRunner {
 	@BeforeClass
 	public static void setUp() {
