diff --git a/lib/Drush/Sql/SqlBase.php b/lib/Drush/Sql/SqlBase.php
index 0d8871ce..e828a8ba 100644
--- a/lib/Drush/Sql/SqlBase.php
+++ b/lib/Drush/Sql/SqlBase.php
@@ -142,7 +142,7 @@ class SqlBase {
    * @return
    *   TRUE on success, FALSE on failure
    */
-  public function query($query, $input_file = NULL, $result_file = '') {
+  public function query($query, $input_file = NULL, $result_file = '', $query_extras = '') {
     $input_file_original = $input_file;
     if ($input_file && drush_file_is_tarball($input_file)) {
       if (drush_shell_exec('gzip -d %s', $input_file)) {
@@ -152,6 +152,9 @@ class SqlBase {
         return drush_set_error(dt('Failed to decompress input file.'));
       }
     }
+    
+    // If no query_extras are passed in the use the default options from cmd or query base.
+    $query_extras = !empty($query_extras) ? $query_extras : drush_get_option('extra', $this->query_extra);
 
     // Save $query to a tmp file if needed. We will redirect it in.
     if (!$input_file) {
@@ -164,7 +167,7 @@ class SqlBase {
       $this->command(),
       $this->creds(),
       $this->silent(), // This removes column header and various helpful things in mysql.
-      drush_get_option('extra', $this->query_extra),
+      $query_extras,
       $this->query_file,
       drush_escapeshellarg($input_file),
     );
diff --git a/lib/Drush/Sql/Sqlpgsql.php b/lib/Drush/Sql/Sqlpgsql.php
index e8945c73..81ceabfe 100644
--- a/lib/Drush/Sql/Sqlpgsql.php
+++ b/lib/Drush/Sql/Sqlpgsql.php
@@ -96,7 +96,9 @@ class Sqlpgsql extends SqlBase {
   }
 
   public function listTables() {
-    $return = $this->query(PSQL_SHOW_TABLES);
+    // listTables should only ever respect the query_extras of the base sql class
+    // and not the cmd extras @see https://github.com/drush-ops/drush/issues/3321
+    $return = $this->query(PSQL_SHOW_TABLES, NULL, '', $this->query_extra);
     $tables = drush_shell_exec_output();
     if (!empty($tables)) {
       return $tables;
