diff --git a/src/Plugin/PasswordConstraint/PasswordStrength.php b/src/Plugin/PasswordConstraint/PasswordStrength.php
index c296552..524c641 100644
--- a/src/Plugin/PasswordConstraint/PasswordStrength.php
+++ b/src/Plugin/PasswordConstraint/PasswordStrength.php
@@ -43,8 +43,9 @@ class PasswordStrength extends PasswordConstraintBase {
     $strength = $password_strength->passwordStrength($password, $userData);
 
     if ($strength['score'] < $configuration['strength_score']) {
-      $validation->setErrorMessage($this->t('The password has a score of @password-score but the policy requires a score of at least @policy-score', array('@password-score'=>$strength['score'], '@policy-score'=>$configuration['strength_score'])));
+	$validation->setErrorMessage($this->t('The password was found in a list of commonly-used passwords. Please choose another.'));
     }
+
     return $validation;
   }
 
@@ -81,7 +82,7 @@ class PasswordStrength extends PasswordConstraintBase {
    * {@inheritdoc}
    */
   public function getSummary() {
-    return $this->t('Password Strength minimum score of @score', array('@score' => $this->configuration['strength_score']));
+	return $this->t('Password must not be overly common.');
   }
+}
 
-}
\ No newline at end of file
